components/Countdown.js

```diff
@@ -0,0 +1,39 @@
+var React = require("react");
+import PropTypes from "prop-types";
+import timediff from "timediff";
+class Countdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.tick = this.tick.bind(this);
+    this.state = { date: props.initialDate, now: Date.now() };
+    this.TimeDiff = this.timeDiff();
+  }
+
+  componentDidMount() {
+    setInterval(this.tick, 1000);
+  }
+  tick() {
+    this.setState({ now: Date.now() });
+    this.TimeDiff = this.timeDiff();
```

Comment:
Why does it needs to be stored in the instance variable BTW? It looks like a state since you’re rendering those values.